kernel/pci/src/lib.rs

```diff
@@ -530,4 +560,77 @@ impl DerefMut for PciDevice {
 pub enum PciConfigSpaceAccessMechanism {
     MemoryMapped = 0,
     IoPort = 1,
-}
\ No newline at end of file
+}
+
+/// A memory-mapped array of [`MsixVectorEntry`]
+pub struct MsixVectorTable {
+    inner: BorrowedSliceMappedPages<MsixVectorEntry, Mutable>,
+}
+
+impl MsixVectorTable {
+    pub fn new(inner: BorrowedSliceMappedPages<MsixVectorEntry, Mutable>) -> Self {
+        Self {
+            inner,
+        }
+    }
+}
+
+impl Deref for MsixVectorTable {
+    type Target = [MsixVectorEntry];
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl DerefMut for MsixVectorTable {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.inner
+    }
+}
+
+/// A single Message Signaled Interrupt entry.
+/// It contains the interrupt number for this vector and the core this interrupt is redirected to.
+#[derive(FromBytes)]
+#[repr(C)]
+pub struct MsixVectorEntry {
+    /// The lower portion of the address for the memory write transaction.
+    /// This part contains the CPU ID which the interrupt will be redirected to.
+    msg_lower_addr:         Volatile<u32>,
+    /// The upper portion of the address for the memory write transaction.
+    msg_upper_addr:         Volatile<u32>,
+    /// The data portion of the msi vector which contains the interrupt number.
+    msg_data:               Volatile<u32>,
+    /// The control portion which contains the interrupt mask bit.
+    vector_control:         Volatile<u32>,
+}
+
+impl MsixVectorEntry {
+    /// Sets interrupt destination & number for this entry
+    pub fn configure(&mut self, cpu_id: CpuId, int_num: InterruptNumber) {
```

Comment:
more descriptive name? "configure" is a bit vague. 

Based on the comments within this function, something like "enable interrupt for cpu" or the like

Reply:
I'd have named it so but there's always a value in there, so it could already be enabled. Should I still name it `enable_...`?

Reply:
"configure" doesn't really say anything about what it does. If it enables something, then yea, i'd vote enable. That implies there should be a corresponding way to disable it.
If it initializes something, you could use init_... too.

Reply:
Simply `init` maybe? since this function simply encodes the cpu ID, the interrupt number and makes sure the interrupt is unmasked.

From the signature:
```rust
MsixVectorEntry::init(CpuId, InterruptNumber);
```
and the new description: Sets interrupt destination & number for this entry and makes sure the interrupt is unmasked (PCI Controller side).

I feel like it's sufficiently clear?

Reply:
sure. I'd need to know more about what the API actually should be for MSI vector entries before I can make an informed decision, but in any case let's just stick with `init` for now i guess.